pkg/routeagent/controllers/route/route.go

```diff
@@ -273,30 +543,56 @@ func (r *Controller) handleRemovedEndpoint(obj interface{}) {
 	if object.Spec.Hostname == hostname {
 		r.cleanRoutes()
 	}
+
+	if object.Spec.ClusterID == r.clusterID {
+		klog.V(6).Infof("Endpoint matches the cluster ID of this cluster")
+		err := r.vxlanDevice.deleteVxLanIface()
+		if err != nil {
+			klog.Errorf("Failed to delete the the vxlan interface on endpoint removal: %v", err)
+			return
+		}
+		r.vxlanDevice = nil
+	}
+
 	klog.V(4).Infof("Removed routes from host")
 }
 
 func (r *Controller) handleRemovedCluster(obj interface{}) {
 	// ideally we should attempt to remove all routes if the endpoint matches our cluster ID
 }
 
+func (r *Controller) handleRemovedPod(obj interface{}) {
+	klog.V(6).Infof("Removing podIP in route controller %v", obj)
+	pod := obj.(*k8sv1.Pod)
+
+	if r.remoteVTEPs.Contains(pod.Status.HostIP) {
+		r.remoteVTEPs.Delete(pod.Status.HostIP)
+		if r.isGatewayNode && r.vxlanDevice != nil {
+			ret := r.vxlanDevice.DelFDB(net.ParseIP(pod.Status.PodIP), "00:00:00:00:00:00")
```

Comment:
Similar as in `processNextPod` - `r.vxlanDevice` could be nil here.

Reply:
As above, please clarify why vxlanDevice would become nil after its created.